.changes/unreleased/Features-20231218-195854.yaml

```diff
@@ -0,0 +1,6 @@
+kind: Features
+body: Move flags from UserConfig in profiles.yml to flags in dbt_project.yml
```

Comment:
Q: do we still support flags in profiles.yml?

Reply:
If no would this mean a potential breaking change?

Reply:
Yes! We fire a deprecation warning if flags are read from profiles.yml, but we still support it: https://github.com/dbt-labs/dbt-core/pull/9998/files#diff-8dc87da2257e24092820e0b29ea50eb16f002dd2c30b5879602b9311941858c0R820-R824

Reply:
Looks like all abstracted under read_project_flags?

Reply:
I'm totally cool with starting to fire this this deprecation warning in v1.7.x

As a general rule, I like the idea of backporting + patching deprecation warnings to actively supported versions, as soon as we decide behavior might change / be officially deprecated (with opt-out) in the next minor release, which could otherwise be multiple months away